source/wp-content/themes/wporg-developer-2023/src/code-related/block.php

```diff
@@ -25,16 +33,95 @@ function init() {
  * @return string Returns the block markup.
  */
 function render() {
+	$uses        = null;
+	$used_by     = null;
+	$has_uses    = false;
+	$has_used_by = false;
+
+	if ( post_type_has_uses_info() ) {
+		$uses     = get_uses();
+		$has_uses = $uses->have_posts();
+	}
+
+	if ( post_type_has_usage_info() ) {
+		$used_by     = get_used_by();
+		$has_used_by = $used_by->have_posts();
+	}
+
+	if ( ! $has_uses && ! $has_used_by ) {
+		return '';
+	}
+
 	$title_block = sprintf(
 		'<h2 class="wp-block-heading">%s</h2>',
 		__( 'Related', 'wporg' )
 	);
 
 	$wrapper_attributes = get_block_wrapper_attributes();
 	return sprintf(
-		'<section %s>%s %s</section>',
+		'<section %s>%s %s %s</section>',
 		$wrapper_attributes,
 		$title_block,
-		'Not Implemented'
+		$has_uses ? do_blocks( get_uses_table( $uses ) ) : '',
+		$has_used_by ? do_blocks( get_used_by_table( $used_by ) ) : '',
 	);
 }
+
+/**
+ * Returns list of rows for the table.
+ *
+ * @param WP_Post[] $posts
+ * @return array[]
+ */
+function get_row_data( $posts ) {
+	$rows = array();
+
+	while ( $posts->have_posts() ) {
+		$posts->the_post();
+		$is_function = ! in_array( get_post_type(), array( 'wp-parser-class', 'wp-parser-hook' ) );
+
+		$rows[] = array(
+			sprintf(
+				'<a href="%s">%s%s</a><code>%s</code></a>',
```

Comment:
Could you use numbered placeholders here?

```suggestion
				'<a href="%1$s">%2$s%3$s</a><code>%4$s</code></a>',
```